concordium-node/src/bin/cli.rs

```diff
@@ -140,6 +142,10 @@ async fn main() -> anyhow::Result<()> {
     )?;
     info!("Consensus layer started");
 
+    // Start stats collecting which depend on querying consensus.
+    let consensus_collector = StatsConsensusCollector::new(consensus.clone())?;
+    node.stats.registry.register(Box::new(consensus_collector))?;
```

Comment:
Is there a real reason for starting the prometheus server before, and then breaking abstraction here to register an additional metric?

Could we not move the prometheus server start here?

Reply:
We are not really breaking the abstraction here, these metrics will just not be enabled until this point.
We could delay setting up the endpoint for scrapping until this point, but I don't really see a point.

Anyway, the challenge is that we need to pass stats to P2PNode to get access in the code related to networking.

Reply:
Ok.